src/parser/MagicServiceQuery.h

```diff
@@ -0,0 +1,69 @@
+// Copyright 2024, University of Freiburg
+// Chair of Algorithms and Data Structures
+// Authors: Johannes Herrmann <johannes.r.herrmann(at)gmail.com>
+//          Christoph Ullinger <ullingec@informatik.uni-freiburg.de>
+
+#pragma once
+
+class SparqlTriple;
+
+namespace parsedQuery {
+
+struct BasicGraphPattern;
+
+class MagicServiceException : public std::exception {
+  std::string message_;
+
+ public:
+  explicit MagicServiceException(const std::string& message)
+      : message_(message) {}
+  const char* what() const noexcept override { return message_.data(); }
+};
```

Comment:
As we don't really catch those types, you can make them easier (all three of them), by just doing
```
class MagicException : public std::runtime_error {
 using std::runtime_error::runtime_error; // Constructors have to be explicitly inherited
};
```
(less boilerplate)